docs: Deprecate CNPStatusUpdates/K8sEventHandover

marseel opened the pull request:
CNP Node status updates are known to cause scalability-related issues. Deprecate it in preparation to remove support in a future release, to minimize the maintenance burden of this code.

Related https://github.com/cilium/cilium/issues/24503

```release-note
Deprecate CNP Node status updates. 
```

squeed reviewed (changes requested):
One last request (sorry) -- can you add something in [upgrade.rst](https://github.com/cilium/cilium/blob/master/Documentation/operations/upgrade.rst) about the deprecation?

marseel committed:
docs: Deprecate CNPStatusUpdates/K8sEventHandover

CNP Status Updates are known to cause scalability-related issues.
Deprecate it in preparation to remove support in a future release,
to minimize the maintenance burden of this code.

Signed-off-by: Marcel Zieba <marcel.zieba@isovalent.com>

qmonnet commented on Documentation/operations/upgrade.rst:
```diff
@@ -351,6 +351,8 @@ Deprecated Options
   ``tunnel=disabled``). To configure the tunneling protocol, set
   ``tunnel-protocol=geneve`` (previously ``tunnel=geneve``).
+* The ``disable-cnp-status-updates``, ``cnp-node-status-gc-interval duration`` and ``enable-k8s-event-handover``
+  options are deprecated and will be removed in v1.15.
```
What are they replaced with or what should users use instead? (You explain the motivation in the commit description, but I wouldn't expect users of this section to check it out.)

qmonnet reviewed (approved):
Thank you!
